src/App.tsx

```diff
@@ -1,14 +1,64 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Dropdown } from './components/Dropdown/Dropdown';
+import { Person } from './types/Person';
 
-export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+const debounce = (callback: (state: string) => void, delay: number) => {
+  let timeoutId: number;
+
+  return (...args: string[]) => {
+    clearTimeout(timeoutId);
+
+    timeoutId = setTimeout(callback, delay, ...args);
+  };
+};
+
+type Props = {
+  delay?: number,
+};
+
+export const App: React.FC<Props> = ({ delay = 1000 }) => {
+  const [selectedPerson, setSelectedPerson] = useState<null | Person>(null);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [isApplied, setIsApplied] = useState(false);
```

Comment:
Use more descriptive names
```suggestion
  const [isSuggestionsActive, setIsSuggestionsActive] = useState(false);
```